src/infrastructure/implementations/blockstack-crux-user-repository.ts

```diff
@@ -120,7 +132,19 @@ export class BlockstackCruxUserRepository implements ICruxUserRepository {
                 cruxUserData = dereferencedCruxpayObject.cruxUserData;
             }
         } else if (cruxUserInformation.registrationStatus.status === SubdomainRegistrationStatus.PENDING) {
-            const cruxpayJson = await this.getCruxpayObjectAndPubKey(cruxID, undefined, publicKeyToAddress(await keyManager.getPubKey()));
+            let cruxpayJson: ICruxpayObjectAndPubKey;
+            try {
+                cruxpayJson = await this.getCruxpayObjectAndPubKey(cruxID, undefined, publicKeyToAddress(cruxpayPubKey));
+            } catch (error) {
+                if (error instanceof PackageError && error.errorCode === PackageErrorCode.GaiaEmptyResponse) {
```

Comment:
can this error be handled inside `getCruxpayObjectAndPubKey` to avoid repetition?

Reply:
we don't have access to the public key there!